tests/sources/helpers/rest_client/test_paginators.py

```diff
@@ -317,3 +371,60 @@ def test_maximum_page(self):
         paginator.update_state(response)  # Page 2
         assert paginator.current_value == 3
         assert paginator.has_next_page is False
+
+    def test_client_pagination_one_based(self, rest_client):
+        pages_iter = rest_client.paginate(
+            "/posts",
+            paginator=PageNumberPaginator(base_page=1, page=1, total_path="total_pages"),
+        )
+
+        pages = list(pages_iter)
+
+        assert_pagination(pages)
+
+    def test_client_pagination_one_based_default_page(self, rest_client):
+        pages_iter = rest_client.paginate(
+            "/posts",
+            paginator=PageNumberPaginator(base_page=1, total_path="total_pages"),
+        )
+
+        pages = list(pages_iter)
+
+        assert_pagination(pages)
+
+    def test_client_pagination_zero_based(self, rest_client):
+        pages_iter = rest_client.paginate(
+            "/posts_zero_based",
+            paginator=PageNumberPaginator(base_page=0, page=0, total_path="total_pages"),
+        )
+
+        pages = list(pages_iter)
+
+        assert_pagination(pages)
+
+
+@pytest.mark.usefixtures("mock_api_server")
+class TestJSONResponseCursorPaginator:
+    def test_update_state(self):
+        paginator = JSONResponseCursorPaginator(cursor_path="next_cursor")
+        response = Mock(Response, json=lambda: {"next_cursor": "cursor-2", "results": []})
+        paginator.update_state(response)
+        assert paginator._next_reference == "cursor-2"
+        assert paginator.has_next_page is True
+
+    def test_update_request(self):
+        paginator = JSONResponseCursorPaginator(cursor_path="next_cursor")
+        paginator._next_reference = "cursor-2"
+        request = Request(method="GET", url="http://example.com/api/resource")
+        paginator.update_request(request)
+        assert request.params["cursor"] == "cursor-2"
```

Comment:
I was surprised that this method modifies the argument by reference without returning it. Could we make it return the modified object?

```suggestion
        updated_request = paginator.update_request(request)
        assert updated_request.params["cursor"] == "cursor-2"
```